### PR TITLE
Sync to upstream 2020-12-07 - easy non-controversial stuff

### PR DESCRIPTION
**Pull request by amilcarlucas**

### CODE REVIEW

**Comment by hamishwillee** [comment]
This all looks non-controversial to me. Most of it comes from direct discussion with ArduPilot team either on PRs or devcall.

File: message_definitions/v1.0/ardupilotmega.xml [hunk 1]
```diff
@@ -27,17 +27,7 @@
     </enum>
     <!-- ardupilot specific MAV_CMD_* commands -->
     <enum name="MAV_CMD">
-      <!-- 211 used by common.xml -->
-      <entry value="212" name="MAV_CMD_DO_AUTOTUNE_ENABLE" hasLocation="false" isDestination="false">
```
Comment (peterbarker) [inline message_definitions/v1.0/ardupilotmega.xml:31]: I don't see PX4 Firmware using this command.

Is there a second user of this command which warrants it being in common?

**Review by peterbarker — CHANGES_REQUESTED** [review]
Could we pop out the autotune_enable patch, please?

This is good apart from that.

Two reasons for the autotune thing; the first is that I don't see another implementation.

The second is that @tridge has muttered something about rejigging autotune for Plane in ArduPilot and I'm not sure how that will work (so we might end up not implementing this message ourselves!)

Not to say that we won't take the patch - just it should be done separately.

**Comment by hamishwillee** [comment]
> Could we pop out the autotune_enable patch, please?

FYI PX4 moved it because they want to have an implementation, and we assumed that ArduPilot already do. Ie not to reinvent the wheel.

**Comment by peterbarker** [comment]
On Tue, 8 Dec 2020, Hamish Willee wrote:

>       Could we pop out the autotune_enable patch, please?
> 
> FYI PX4 moved it because they want to have an implementation, and we assumed that ArduPilot already do. Ie not to reinvent the wheel.

Yes, we have an implementation - but only in ArduPlane.  And that's slated 
for change in ways yet to be determined.  I've never heard of anyone 
actually using this mavlink message (or mission command equivalent) - we 
have a dedicated mode for autotune which works just like fly-by-wire-A 
except for monitoring the command responses.

Not saying this patch won't be merged, but it should go to a DevCall (I 
won't merge it off my own bat).

**Comment by amilcarlucas** [comment]
Done.

**Review by tridge — APPROVED** [review]

**Pull request merged**